### PR TITLE
Fix smoking taking time to not work.

### PR DESCRIPTION
**Pull request by Ramza13**
#### Summary
None

#### Purpose of change
To fix usage of a joint or other smokable items from taking time before telling you you don't have the right tools.

#### Describe the solution
Do the check for if you have the tools before allowing you to start.

### CODE REVIEW

File: src/iuse.cpp [hunk 1]
```diff
@@ -453,17 +453,21 @@ void remove_radio_mod( item &it, Character &p )
...
+cata::optional<std::string> iuse::can_smoke( const player &u )
 {
     auto cigs = u.items_with( []( const item & it ) {
         return it.active && it.has_flag( flag_LITCIG );
     } );
-    if( cigs.empty() ) {
-        return true;
+
+    if( !cigs.empty() ) {
+        return _( "You're already smoking a %s!" ), cigs[0]->tname();
```
Comment (BrettDong) [inline src/iuse.cpp:464]:
```suggestion
        return string_format( _( "You're already smoking a %s!" ), cigs[0]->tname() );
```

Comment (Ramza13) [reply src/iuse.cpp:464]: Doesn't _() do that already?

Comment (BrettDong) [reply src/iuse.cpp:464]: I think you are returning a comma expression now.

Comment (anothersimulacrum) [reply src/iuse.cpp]: If it did, the arguments would need to be inside `_()`. I'm surprised nothing is complaining about this.

**Pull request merged**